### PR TITLE
[fabric] Introduce helm chart deployment capability for version 2.5

### LINKED ISSUE #2484: epic: platform deployments using just helm commands and charts
**As Developers, we want to implement deployment of platforms using just helm charts**

**Few of the key improvements in the platform's charts folder as follows:**
- Update `parameters table` to accurately reflect variable hierarchy and use `camelCase` for parameter/variable names.
- Align the READMEs of a specific platform with the structure and design of `BESU` READMEs.
- Implement hook concept to delete all the supported resources of a chart while doing the network reset
- Utilize `helpers.tpl` to assign the deployment name instead of metadata section in template (.tpl) and values.yaml files.
- Organize common configuration variables, like those associated with Vault and Cluster, under the `global` section in template (.tpl) and values.yaml files. Ensure these variables are used with the updated hierarchy within the respective chart.
- Implement chart dependencies and  use pre install hook concept to create prerequisite Kubernetes  resources and run one time tasks  

**Required testing**:
- Perform Helm lint manually to quickly validate the chart, followed by the successful network deployment and reset using just helm charts and commands.
- For any concerns and their solutions, please follow the Besu Platform.

**This issue needs to be fixed for the following platforms**:
- [x] Besu @sownak 
- [x] Quorum @saurabhkumarkardam 
- [x] Fabric @mgCepeda 
- [x] R3-Corda-Ent @suvajit-sarkar 
- [x] R3-Corda @sownak 
- [x] Substrate @saurabhkumarkardam and @saikumarbommakanti

### PR DESCRIPTION
**Pull request by mgCepeda**
**Primary Changes**

1. This PR includes changes to deploy fabric 2.5.4 without a channel using helm charts.
2. Version 2.2.2 is pending
3. Deploy with Ansible pending
4. Update README.md files is pending

**Changes in charts**
platforms/hyperledger-fabric/charts/fabric-ca-server 
platforms/hyperledger-fabric/charts/fabric-cacerts-gen 
platforms/hyperledger-fabric/charts/fabric-catools 
platforms/hyperledger-fabric/charts/fabric-cli
platforms/hyperledger-fabric/charts/fabric-orderernode 
platforms/hyperledger-fabric/charts/fabric-peernode

fixes #2484

### CODE REVIEW

**Review by sownak — CHANGES_REQUESTED** [review]

File: platforms/hyperledger-fabric/charts/README.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+### To setup Orderer organization
+```bash
+kubectl create namespace supplychain-net 
+
+helm install ca-certs ./fabric-cacerts-gen --namespace supplychain-net --values ./values/noproxy-and-novault/ordererOrganization/cacerts-gen.yaml
+
+helm install supplychain-server ./fabric-ca-server --namespace supplychain-net --values ./values/noproxy-and-novault/ordererOrganization/ca-server.yaml
+
+helm install supplychain-tools ./fabric-catools --namespace supplychain-net --values ./values/noproxy-and-novault/ordererOrganization/ca-tools-orderer.yaml
```
[the remark this answers is not in the document]
Comment (mgCepeda) [reply platforms/hyperledger-fabric/charts/README.md:55]: Done

File: platforms/hyperledger-fabric/charts/fabric-ca-server/values.yaml [hunk 2]
```diff
@@ -3,111 +3,89 @@
...
-annotations:
-  # Extra annotations for the service
-  service: []
-  # Extra annotations for the PVC
-  pvc: []
-  
+      #Eg. clusterIpPort: 7054
+      clusterIpPort: 7054
+ 
 proxy:
```
[the remark this answers is not in the document]
Comment (mgCepeda) [reply platforms/hyperledger-fabric/charts/fabric-ca-server/values.yaml:85]: Done

File: platforms/hyperledger-fabric/charts/fabric-ca-server/values.yaml [hunk 3]
```diff
@@ -3,111 +3,89 @@
...
+  #Provide the size for CA
+  #Eg. size: 512Mi
+  size: 512Mi
 
 service:
   #Provide service type for the pod
-  #Eg. servicetype: NodePort
-  servicetype: ClusterIP
+  #Eg. serviceType: NodePort
+  serviceType: ClusterIP
```
[the remark this answers is not in the document]
Comment (mgCepeda) [reply platforms/hyperledger-fabric/charts/fabric-ca-server/values.yaml:75]: Done

File: platforms/hyperledger-fabric/charts/fabric-ca-server/values.yaml [hunk 4]
```diff
@@ -3,111 +3,89 @@
...
-  annotations: 
+labels:
+  service: []
+  pvc: []
+  deployment: []
 
 server:
   #Provide name for ca server deployment
   #Eg. name: ca
   name: ca
```
[the remark this answers is not in the document]
Comment (mgCepeda) [reply platforms/hyperledger-fabric/charts/fabric-ca-server/values.yaml:56]: Done

File: platforms/hyperledger-fabric/charts/fabric-ca-server/templates/configmap.yaml [hunk 5]
```diff
@@ -9,13 +9,14 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   name: {{ $.Values.server.name }}-config
-  namespace: {{ $.Values.metadata.namespace }}
+  namespace: {{ .Release.Namespace }}
   labels:
-    app.kubernetes.io/name: {{ $.Values.server.name }}-config
-    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    app.kubernetes.io/name: {{ $.Values.server.name }}
```
[the remark this answers is not in the document]
Comment (mgCepeda) [reply platforms/hyperledger-fabric/charts/fabric-ca-server/templates/configmap.yaml:14]: Done

File: .gitignore [hunk 6]
```diff
@@ -41,4 +41,5 @@
 **/charts/*.tgz
 **/files/*.json
 **/files/*.crt
+**/files/*.yaml
```
[the remark this answers is not in the document]
Comment (mgCepeda) [reply .gitignore:44]: Done

File: platforms/hyperledger-fabric/charts/fabric-ca-server/README.md [hunk 7]
```diff
@@ -139,7 +139,7 @@ The [values.yaml](https://github.com/hyperledger/bevel/blob/develop/platforms/hy
 | ----------------------| -------------------------------------------------------------------------|--------------------------------|
 | provider              | Proxy/ingress provider. Possible values: "haproxy" or "none"             | haproxy                        |
 | type                  | Type of the deployment. Possible values: "orderer", "peer", or "test"    | test                           |
-| external_url_suffix   | External URL suffix for the organization                                 | org1proxy.blockchaincloudpoc.com    |
+| externalUrlSuffix   | External URL suffix for the organization                                 | org1proxy.blockchaincloudpoc.com    |
```
[the remark this answers is not in the document]
Comment (mgCepeda) [reply platforms/hyperledger-fabric/charts/fabric-ca-server/README.md:142]: Done. Only the name of the variable was changed, not the value.

**Review by sownak — APPROVED** [review]

**Review by suvajit-sarkar — APPROVED** [review]
LGTM

**Pull request merged**